### PR TITLE
rpc.statd is required for file locking in NFSv3 mounts

### LINKED ISSUE #4953: [convoy-nfs] requires nolock mount option.. no nfs locking in convoy nfs
Mounting convoy-nfs fails unless nolock is included in the mount options. It appears that while rpcbind is started by the launch script, the rpc.statd is not. File locking over nfs is thus unavailable.

We actually need locking over nfs.

https://github.com/rancher/convoy-agent/blob/master/packaging/convoy-agent/launch#L171

 5/11/2016 9:46:04 PMWaiting for metadata
    5/11/2016 9:46:04 PMtime="2016-05-12T04:46:04Z" level=info msg="Execing [/usr/bin/nsenter --mount=/proc/21656/ns/mnt -F -- /var/lib/docker/aufs/mnt/9362b696c64335f026344abfdbf1ba74329ff7e039ceb8828de7041c31f6aa8f/var/lib/rancher/convoy-agent/share-mnt --stage2 /var/lib/rancher/convoy/convoy-nfs-47f99cab-b92d-45c2-88cb-e086cad2775a -- /launch volume-agent-nfs-internal 21656]"
    5/11/2016 9:46:04 PMRegistering convoy socket at /var/run/convoy-convoy-nfs.sock
    5/11/2016 9:46:04 PMMounting at: /var/lib/rancher/convoy/convoy-nfs-47f99cab-b92d-45c2-88cb-e086cad2775a/mnt
    5/11/2016 9:46:04 PMMounting nfs. Command: mount -t nfs -o rsize=32768,wsize=32768,noatime,intr XX.XX.XX.XX:/export/rdata/nsfdata /var/lib/rancher/convoy/convoy-nfs-47f99cab-b92d-45c2-88cb-e086cad2775a/mnt
    5/11/2016 9:46:04 PMmount.nfs: rpc.statd is not running but is required for remote locking.
    5/11/2016 9:46:04 PMmount.nfs: Either use '-o nolock' to keep locks local, or start statd.
    5/11/2016 9:46:04 PMmount.nfs: an incorrect mount option was specified

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v1.0.1` Cattle: `v0.159.7` UI: `v1.0.5` |
|Access|`localauth` `admin`|
|Route|`environments.index`|

### PR DESCRIPTION
**Pull request by janeczku**
Follow up to fix https://github.com/rancher/rancher/issues/4953

### CODE REVIEW

File: packaging/convoy-agent/launch [hunk 1]
```diff
@@ -213,7 +214,8 @@ volume_agent_efs_internal() {
     echo "unix://$CONVOY_SOCK_ON_HOST" > /etc/docker/plugins/$STACK_NAME.spec
 
     echo "Mounting at: $MNT_PT"
-    rpcbind
+    nsenter -t $TARGET_PID -n rpcbind
```
Comment (cjellick) [inline packaging/convoy-agent/launch:217]: @janeczku Does rpcbind or rpc.statd even need to run for efs?

Comment (janeczku) [reply packaging/convoy-agent/launch:217]: Looks like EFS supports NFS v4.x only - so no. I'll update the patch.

Comment (janeczku) [reply packaging/convoy-agent/launch:217]: > While Amazon Elastic File System does not support NFSv2, or NFSv3, Amazon EFS supports both NFSv4.1 and NFSv4.0, except for the following features.

http://docs.aws.amazon.com/efs/latest/ug/nfs4-unsupported-features.html

> NFSv4 does not use rpcbind as the NFS server itself listens on TCP port 2049 for service requests. The mounting and locking protocols are also integrated into the NFSv4 protocol, so the lockd and rpc.statd services are also not required. These refinements mean that firewall configuration for NFSv4 is no more difficult than for a service such as HTTP.

https://docs.oracle.com/cd/E37670_01/E41138/html/ol_about_nfs.html

**Comment by janeczku** [comment]
Old kernels need rpcbind even for NFSv4
http://serverfault.com/a/639796

**Comment by cjellick** [comment]
LGTM

**Pull request merged**